### PR TITLE
feat(query): migrate func to func-v2

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

```
lpad
rpad
replace
strcmp
locate， POSITION(substr IN str) is a synonym for LOCATE(substr,str).
instr alias locate
```

Fixes part of #6766

### CODE REVIEW

File: src/query/functions-v2/src/scalars/string.rs [hunk 1]
```diff
@@ -106,6 +120,346 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+            };
+            match res {
+                Ordering::Equal => 0,
+                Ordering::Greater => 1,
+                Ordering::Less => -1,
+            }
+        },
+    );
+
+    // registry.register_4_arg::<StringType, NumberType<i64>, NumberType<i64>, StringType, StringType, _, _>(
```
Comment (TCeason) [inline src/query/functions-v2/src/scalars/string.rs:228]: Add 4_arg function. I'm not sure about the number of functions which has 4 args. If not too much I think maybe we just need to use register_function_factory impl the function?@andylokandy

File: src/query/expression/src/function.rs [hunk 2]
```diff
@@ -509,6 +509,182 @@ impl FunctionRegistry {
             }));
     }
 
+    pub fn register_4_arg<I1: ArgType, I2: ArgType, I3: ArgType, I4: ArgType, O: ArgType, F, G>(
```
Comment (TCeason) [inline src/query/expression/src/function.rs:512]: That's so crazy.

Comment (andylokandy) [reply src/query/expression/src/function.rs:512]: OMG

**Comment by sundy-li** [comment]
We should add macro to represent `register_x_arg `

**Comment by andylokandy** [comment]
@sundy-li i'll take the task. so, @TCeason, would you like to revert to register_x_args version?

**Comment by TCeason** [comment]
> @sundy-li i'll take the task. so, @TCeason, would you like to revert to register_x_args version?

Ok, I will wait your pr

**Review by sundy-li — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]
+1

**Pull request merged**